### ISSUE TITLE
Semantic annotations / triplification

### ISSUE THREAD

**Issue by benjelloun** [mlcommons/croissant#739]
Define a complete mechanism to add semantic annotations to Croissant data, that builds on the existing dataType/equivalentProperty approach to annotate Croissant data with classes and properties from existing vocabularies. 

This mechanism should make it possible to generate a triple representation of annotated Croissant data, given the user sufficient control over the generated triples.

**Comment by JervenBolleman** [comment]
May I suggest to investigate if prior work like [csv on the web](https://csvw.org/), [r2rml](https://www.w3.org/TR/r2rml/)/[rml](https://rml.io/specs/rml/) can be used/extended or inspire this feature.

**Comment by benjelloun** [comment]
Absolutely! I for one am very familiar with these approaches, and will definitely look at how we can reuse from them.
